src/internal/FluencePeer.ts

```diff
@@ -238,7 +238,11 @@ export class FluencePeer {
              * Initiates a new particle execution starting from local peer
              * @param particle - particle to start execution of
              */
-            initiateParticle: (particle: Particle): void => {
+            initiateParticle: (particle: Particle, onStageChange: (stage: ParticleExecutionStage) => void): void => {
+                if (!this.getStatus().isInitialized) {
+                    throw 'Cannon initiate new particle: peer is no initialized';
```

Comment:
Will this error fail the promise returned by `callFunction`?

Reply:
yes

Reply:
Let's fail the promise with this error.

Reply:
I don't understand you. This function does not return promises. It's synchronous. It is not aware of the context it is used in. The promise constructed by `callFunction` will get rejected if this error is thrown

Reply:
I see, `new Promise` body has an invisible `try catch`. Ok, then.